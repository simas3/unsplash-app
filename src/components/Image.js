import React from 'react';

const Image = props =>
    <div className="image-item">
        <a href={props.link}>
            <img className="image" src={props.url} alt="" />
        </a>
    </div>;

export default Image;