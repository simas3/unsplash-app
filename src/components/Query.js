import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { deleteQuery, getImages } from '../actions/imageActions'
import { connect } from 'react-redux'

export class Query extends Component {


    onQueryClick() {
        this.props.getImages(this.props.query)
    }

    onRemove() {
        this.props.deleteQuery(this.props.count - 1)
    }

    render() {
        console.log(this.props)
        return (
            <div className="queryoption">
                <button className="querybutton querybutton--link" onClick={this.onQueryClick.bind(this)} >{this.props.count}. {this.props.query}</button>
                <button className="querybutton querybutton--link" onClick={this.onRemove.bind(this)}>x</button>
            </div>
        )
    }
}

Query.propTypes = {
    deleteQuery: PropTypes.func.isRequired,
    getImages: PropTypes.func.isRequired
}

export default connect(null, { deleteQuery, getImages })(Query)
