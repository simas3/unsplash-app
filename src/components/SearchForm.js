import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { getImages, saveQuery } from '../actions/imageActions'

export class SearchForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            query: '',
            error: ''
        }

        this.onChange = this.onChange.bind(this)
        this.onSearch = this.onSearch.bind(this)
        this.onSave = this.onSave.bind(this)
    }

    onChange(e) {
        this.setState({ [e.target.name]: e.target.value })
    }

    onSearch(e) {
        e.preventDefault()
        if (!this.state.query) {
            this.setState(() => ({ error: 'Please Enter Your Search Query' }))
        } else {
            this.props.getImages(this.state.query)
            this.setState({ error: '', query: '' })
        }

    }

    onSave(e) {
        e.preventDefault()
        if (!this.state.query) {
            this.setState(() => ({ error: 'Can\'t save empty query' }))
        } else {
            this.props.saveQuery(this.state.query)
            this.setState({ error: '', query: '' })
        }
    }

    render() {
        return (
            <div>
                <form className="query" onSubmit={this.onSearch}>
                    <input
                        className="query__input"
                        placeholder="Search images..."
                        name="query"
                        value={this.state.query}
                        onChange={this.onChange}
                    />
                    <button type="submit" className="button">Search</button>
                    <button type="button" className="button" onClick={this.onSave}>Save</button>
                </form>
                {this.state.error && <p className="query-error">{this.state.error}</p>}
            </div>
        )
    }
}

SearchForm.propTypes = {
    getImages: PropTypes.func.isRequired,
    saveQuery: PropTypes.func.isRequired
}

export default connect(null, { getImages, saveQuery })(SearchForm)
