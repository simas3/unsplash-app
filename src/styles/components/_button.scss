
.button{
    background: $purple;
    border: none;
    border-bottom: .3rem solid darken($purple,10%);
    color: white;
    font-weight:500;
    padding: $ssize;
    margin-right: 20px;
    width: 12%;
    text-align: center;
}




@media (max-width: 60rem) {
    .button {
       margin: 0;
       width: 100%;

    }
 }
 
 