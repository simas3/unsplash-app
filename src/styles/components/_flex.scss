.flex-container  {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    max-width: 1600px;
    justify-content: space-between;
    margin: 0 auto;
}

.flex-images {
    flex-basis: 100%
}


@media (max-width: 1200px) {
    .flex-container  {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        max-width: 1200px;
        justify-content: center;
        margin: 0 auto;
    }

    .flex-images {
        flex-basis: 1600px
    }
}