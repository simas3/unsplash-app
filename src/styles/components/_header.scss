


.header {
    text-align: center;
    background: $off-black;
    margin-bottom: $ssize;
    color: white;
    padding: $msize 0;
}

.header__title {
    font-size: $xlsize;
    margin: 0 0 20px 0;
}

.header__subtitle {
    color: $off-white;
    font-size: $msize;
    font-weight: 500;
    margin: 0;
}


@media (min-width: 45rem) {
    .header {
        margin-bottom: $lsize;
    }
 
 
} 
