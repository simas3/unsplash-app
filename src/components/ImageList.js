import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { ClipLoader } from 'react-spinners';
import Image from './Image'

export class ImageList extends Component {

    render() {

        const { images, loading, total } = this.props.images

        let imagesContent;

        if (loading) {
            imagesContent = (
                <div className="spinner">
                    <ClipLoader
                        sizeUnit={"px"}
                        size={100}
                        color={'#8357c5;'}
                        loading={loading}
                    />
                </div>
            )
        } else if (typeof images != "undefined" && images != null && images.length != null && images.length > 0) {
            console.log(images)
            imagesContent = (
                <div className="images-grid">
                    {images.map(image => (
                        <Image
                            url={image.urls.thumb}
                            link={image.links.html}
                            key={image.id}
                        />
                    ))}
                </div>
            )
        } else if (total !== undefined && images.length === 0) {
            imagesContent = (
                <div className="query-error">
                    <p>There was no images found</p>
                </div>
            )
        }

        return (
            <div className="flex-images">
                {imagesContent}
            </div>
        )
    } q
}

ImageList.propTypes = {
    images: PropTypes.object.isRequired
}

const mapStateToProps = (state) => ({
    images: state.images
})

export default connect(mapStateToProps)(ImageList)
