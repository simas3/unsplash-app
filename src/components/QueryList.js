import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Query from './Query'
import { deleteQueries } from '../actions/imageActions'

export class QueryList extends Component {

    deleteQueries() {
        this.props.deleteQueries()
    }

    render() {
        const { queryHistory } = this.props.images

        return (
            <div className="flex-queries">
                {queryHistory.length > 0 ? (
                    <div className="querylist">
                        <div className="querylist-header">
                            <h3 className="querylist-header__title">Your Saved Queries</h3>
                            <button className="querybutton querybutton--link" onClick={this.deleteQueries.bind(this)}>X</button>
                        </div>
                        {
                            queryHistory.map((query, index) => {
                                return <Query key={index} query={query} count={index + 1} />
                            })
                        }
                    </div>) : null}
            </div>
        )
    }
}

QueryList.propTypes = {
    deleteQueries: PropTypes.func.isRequired,
    images: PropTypes.object.isRequired
}

const mapStateToProps = (state) => ({
    images: state.images
})


export default connect(mapStateToProps, { deleteQueries })(QueryList)
