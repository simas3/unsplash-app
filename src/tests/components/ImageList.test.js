import React from 'react'
import { ImageList } from '../../components/ImageList'
import { configure, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('should render ImageList', () => {
    const wrapper = shallow(<ImageList images={{}} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})