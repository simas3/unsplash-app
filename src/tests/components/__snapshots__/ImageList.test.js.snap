// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ImageList 1`] = `
<div
  className="flex-images"
/>
`;
